components/clipboard-button/index.js

```diff
@@ -77,12 +77,12 @@ class ClipboardButton extends Component {
 	render() {
 		// Disable reason: Exclude from spread props passed to Button
 		// eslint-disable-next-line no-unused-vars
-		const { className, children, onCopy, onFinishCopy, text, ...buttonProps } = this.props;
+		const { className, children, onCopy, onFinishCopy, text, icon = 'admin-links', ...buttonProps } = this.props;
 		const classes = classnames( 'components-clipboard-button', className );
 
 		return (
 			<span ref={ this.bindContainer }>
-				<IconButton { ...buttonProps } icon="admin-links" className={ classes }>
+				<IconButton { ...buttonProps } icon={ icon } className={ classes }>
```

Comment:
Starting to think we should leverage a render prop or children to avoid all these props which has nothing to do with the "Clipboard" Button. By this I mean doing something like this here: `<span>{render( { bindButton } ) }</span>` .

Which means the Clipboard button would not render the button itself but delegates to the caller. its role would be to just add the behavior.

I believe it could be quite complex to achieve. As a quick way forward, I propose just doing something like:

`Tag = icon ? 'IconButton' : 'Button';`

Also let's make the icon empty by default to avoid having to pass `icon={ null }`